packages/yoga/src/Heading/web/BackButton.jsx

```diff
@@ -1,26 +1,78 @@
 import React from 'react';
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 import PropTypes from 'prop-types';
 import { ArrowLeft } from '@gympass/yoga-icons';
 import Button from '../../Button';
+import Box from '../../Box';
+import Icon from '../../Icon';
 
-const ButtonIcon = styled(Button.Icon)`
+const ButtonIcon = styled(Button)`
```

Comment:
There are other components that will have the same change in `Button.Icon`, such as `Drawer`, shouldn't this style be a new variant of `Button.Icon` so it can be reused?

FYI @eduardobreno @rafaelcoleta

Reply:
Maybe, only implementing the possibility to pass a `children` to `Button.Icon`, and using `icon` from props or `children` will be sufficient here. What do you think? 

Because most of the styles I added for the `buttons` on the `Heading` component, were just to make the `button` have some style like the `Button.Icon`.

Reply:
I think it will make it easier to reuse if this is part of the `Button.Icon` spec, the change looks like one of the variants in the "Usage" section, with a change in colors.
https://gympass.github.io/yoga/components/button/icon#usage

Reply:
Hey team! This is a good suggestion, and I'm on board that this is something that we should do. But maybe this is not the moment.

I think that in this first moment, we can introduce this duplication as a technical debt and refactor this in the future.

Our main concern is not the code, but our documentation and Figma (for designers). We would create a new variation to the `Button` component, and this needs to have the necessary attention.

Does this make sense?

Reply:
That's fine, but in the future we will have rework in at least 3 different usage scenarios.